profiling/src/exporter/mod.rs

```diff
@@ -130,30 +145,55 @@ impl ProfileExporter {
     ) -> anyhow::Result<Request> {
         let mut form = multipart::Form::default();
 
-        form.add_text("version", "3");
-        form.add_text("start", start.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string());
-        form.add_text("end", end.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string());
-        form.add_text("family", self.family.as_ref());
-
-        for tags in self.tags.as_ref().iter().chain(additional_tags.iter()) {
-            for tag in tags.iter() {
-                form.add_text("tags[]", tag.to_string());
-            }
+        // combine tags and additional_tags
+        let mut tags_profiler = String::new();
+        let other_tags = additional_tags.into_iter();
+        for tag in self.tags.iter().chain(other_tags).flatten() {
+            tags_profiler.push_str(tag.as_ref());
+            tags_profiler.push(',');
         }
+        tags_profiler.pop(); // clean up the trailing comma
+
+        let attachments: Vec<String> = files.iter().map(|file| file.name.to_owned()).collect();
+
+        let event = json!({
+            "attachments": attachments,
+            "tags_profiler": tags_profiler,
+            "start": start.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string(),
+            "end": end.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string(),
+            "family": self.family.as_ref(),
+            "version": "4",
+        })
+        .to_string();
+
+        form.add_reader_file_with_mime(
+            // intake will look for a field of this name
+            "event",
+            // this one shouldn't be compressed
+            Cursor::new(event),
+            // intake doesn't care about filename in this case but RFC uses event.json
+            "event.json",
+            mime::APPLICATION_JSON,
+        );
 
         for file in files {
-            form.add_reader_file(
-                format!("data[{}]", file.name),
-                Cursor::new(file.bytes.to_owned()),
-                file.name,
-            )
+            let mut encoder = FrameEncoder::new(Vec::new());
+            encoder.write_all(file.bytes)?;
+            let encoded = encoder.finish()?;
+            // intake does not care about these name of the form field for these
+            form.add_reader_file(file.name, Cursor::new(encoded), file.name)
```

Comment:
I'm not quite sure this comment adds a lot -- we're sending the file name as the file name which is kinda the obvious thing to do...? (So maybe remove it?)

Reply:
At least currently, the code doesn't match the "examples" from confluence and the RFC, so that's why I left a comment. I will improve it.